### PR TITLE
Support binary/varbinary type

### PR DESCRIPTION
**Pull request by iceTTTT**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #832 #7319 #6615 

## What this PR does / why we need it:
Support binary/varbinary type in mo(disaplay, basic conversion).
Change the implementation of binary function.
Add/change some bvt cases related to binary type.
Support Aggr func max/min/bit_and/bit_or/bit_xor/group_concat/any_value for binary/varbinary data type
Support some operators/builtin functions for binary/varbinary type

### CODE REVIEW

**Review by iamlinjunhong — APPROVED** [review]

**Comment by iceTTTT** [comment]
Tested jdbc insert/select of binary/varbinary type.
Got the same result with mysql.

**Review by ouyuanning — APPROVED** [review]
good job

**Review by daviszhen — APPROVED** [review]

**Review by xzxiong — APPROVED** [review]
lgtm

File: test/distributed/cases/dtype/binary.test [hunk 1]
```diff
@@ -0,0 +1,16 @@
+-- @suite                                                                                                                                                                                
+-- @case
+-- @desc:test for binary datatype
+-- @label:bvt
+
+#Test for basic syntax.
+drop table if exists t1;
+create table t1(a binary(2));
+insert into t1 values("6");
```
Comment (heni02) [inline test/distributed/cases/dtype/binary.test:9]: plz add binary type case about null,out of range ,insert chinese characters etc cases

**Review by reusee — APPROVED** [review]

File: test/distributed/cases/dtype/binary.test (hunk 1 above)
Comment (iceTTTT) [reply test/distributed/cases/dtype/binary.test:15]: Test cases added.

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Review by aressu1985 — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by m-schen — APPROVED** [review]

**Pull request merged**